app/models/apply_application.rb

```diff
@@ -1,14 +1,13 @@
 # frozen_string_literal: true
 
 class ApplyApplication < ApplicationRecord
-  belongs_to :provider
```

Comment:
Thinking out loud, would there be an advantage to retain the belongs to relationship?
```ruby
belongs_to :provider, foreign_key: :provider_code, primary_key: :code
```

so that the job above can potentially be 
```ruby
ApplyApplication.joins(:provider).importable.each do |...
```

Reply:
It would be more efficient. Happy to go along with this.